fish-rust/src/event.rs

```diff
@@ -817,7 +821,9 @@ fn event_fire_delayed_ffi(parser: Pin<&mut parser_t>) {
 /// Enqueue a signal event. Invoked from a signal handler.
 pub fn enqueue_signal(signal: usize) {
     // Beware, we are in a signal handler
-    PENDING_SIGNALS.mark(signal);
+    if let Some(s) = Signal::try_new(signal as i32) {
```

Comment:
This is too much scary code in a signal handler. We should have `mark()` work in primitive types only.